### PR TITLE
Supplying a fix for "#836: vm creation is failing on vsanDatastore"

### PR DESCRIPTION
**Pull request by shuklanirdesh82**
Vm creation was not complaining for vmfs as it doesn't use osfs code path and fails only for vsan as it is objectBased FS and vm-name was appending twice hence the issue.

Testing Done: Yes (both on vsan and vmfs)

vmdk_ops_test.py tests are passing locally and able to create vm successfully.

```
Running unit tests in /tmp/vmdk_ops_unittest22238/vmdk_ops_test.py...
.................
----------------------------------------------------------------------
Ran 17 tests in 272.076s

OK
```
//CC @ashahi1 @shaominchen

### CODE REVIEW

File: esx_service/vmdk_ops_test.py [hunk 1]
```diff
@@ -397,14 +397,11 @@ def create_vm(si, vm_name, datastore):
...
                                                                                        hosts, resource_pool))
-
-        datastore_path = '[' + datastore + '] ' + vm_name
-
         # bare minimum VM shell, no disks. Feel free to edit
         vmx_file = vim.vm.FileInfo(logDirectory=None,
                                    snapshotDirectory=None,
                                    suspendDirectory=None,
-                                   vmPathName=datastore_path)
+                                   vmPathName='[' + datastore + '] ')
```
Comment (msterin) [inline esx_service/vmdk_ops_test.py]: Please rename create_vm param  from `datastore` to `vm_path` and it will be good to go

Comment (shuklanirdesh82) [inline esx_service/vmdk_ops_test.py]: Sure, Thanks!

File: esx_service/vmdk_ops_test.py [hunk 2]
```diff
@@ -479,7 +476,7 @@ def setUp(self):
         si = vmdk_ops.get_si()
         error, self.vm = create_vm(si=si, 
                                    vm_name=self.vm_name, 
-                                   datastore=self.datastore_name)
+                                   vm_path=self.datastore_name)
```
Comment (msterin) [inline esx_service/vmdk_ops_test.py]: this is still weird... why self.datastore_name actually keep the vm_path ? 
I'll approve to unbock , but please investigate as a separate item

Comment (shuklanirdesh82) [inline esx_service/vmdk_ops_test.py]: Actually ```create_vm``` is internally prepares vm_path (vmHome) i.e. ```vm_path = datastore_name + vmDisplayName```.

So I am going to rename it with 'datastore_name' with more meaningful name to reduce the confusion.

**Pull request merged**